### ISSUE TITLE
Allow scanning of multi-page documents

### ISSUE THREAD

**Issue by mvdkleijn** [bauerj/paperless_app#34]
**Is your feature request related to a problem? Please describe.**
Some of the documents I receive are multi-page documents.

**Describe the solution you'd like**
Select add button, select single or multi page, scan each page, dump it to a pdf, upload that to paperless(-ng)

**Describe alternatives you've considered**
Scanner.. but I have neither the space nor the time to use one. It is basically not suited for my workflow.

**Comment by jonaswinkler** [comment]
> Hey Jonas, I'm currently looking into bauerj/paperless_app#34. I was wondering if it would be easy to change the API to allow uploading multiple images to be merged into a single document.
> The alternative would be for us to add some code to create PDF documents

Hm. We'd need to combine multiple image files into one document at some point. Paperless can only store exactly one file per document. PDF is an option. TIFF is another option, that supports multiple pages, but the python imaging library `Pillow` has some issues with certain TIFF formats (I'm investigating that right now).

How difficult would it be to add PDF conversion over here? Not sure about library support. I've got some libraries in paperless already that offer PDF creation and processing, but: I'd also have to write logic for PDF merging in addition to extensions to the API, and all that is not exactly straight foward.

**Comment by bauerj** [comment]
Okay, in that case I will try to implement this client-side. This would also have the advantage of being able to work with the old Paperless.

As for library support, [`pdf`](https://pub.dev/packages/pdf) seems like it should be able to do what we need. Not sure how well this will work though.

**Comment by mvdkleijn** [comment]
Full disclosure from my side (forgot to mention this in my original report): there is a sort of workaround and it goes like this..

1. Scan pages using third party app like CamScanner
2. Convert to PDF using the third party app
3. Share to Paperless app
4. Login to web based Paperless-NG
5. Correct document title and other meta data

While this is sort of do-able, its hardly ideal and requires more tools / apps than I'd like personally. The following flow seems more logical:

1. Scan pages using Paperless app
2. Enter document name
3. Upload to Paperless-NG

Anyway, while nice, there is a workaround.

**Comment by Mattie112** [comment]
Yes this would be great to have. I now make multiple pictures on my phone then and then combine them on the PC (yes the probably is an app for that). But an option in the app would be great!

**Comment by jcobreros** [comment]
I would also love this feature. Right now I'm using Google drive scanner / Office Lens for multi page document scanning, then I share with Paperless. But I would like to only save the document in one place.

**Comment by donbing** [comment]
same here.. even just an option to merge several documents into one after upload would help.

**Comment by shkm** [comment]
For anyone looking for other options: I'm using Genius Scan. I've set up an FTP server around paperless' watch folder and set that destination for auto-export in Genius Scan. It's a very smooth workflow.

**Comment by dereitz** [comment]
I don't see a way to vote for this but +1 as this would be a great feature!

**Comment by mvdkleijn** [comment]
> I don't see a way to vote for this but +1 as this would be a great feature!

Generally you do a thumbs up on the original issue description or comment if you want to do a +1... instead of a comment. :-) (for future reference)
